Composer/plugins/samples/src/index.ts

```diff
@@ -0,0 +1,120 @@
+    tags: ["Basic"],
+    support: ["C#", "JS"]
+  },
+  EchoBot: {
+    name: 'Echo Bot',
+    description: 'A bot that echoes and responds with whatever message the user entered',
+    index: 1,
+  },
+  EmptyBot: {
+    name: 'Empty Bot',
+    description: 'Basic bot template that is ready for your creativity',
+    index: 2,
+  },
+  TodoSample: {
+    name: 'Simple Todo',
+    description: 'A sample bot that allows you to add, list, and remove to do items.',
+    index: 3,
+  },
+  ToDoBotWithLuisSample: {
+    name: 'Todo with LUIS',
+    description: 'A sample bot that allows you to add, list, and remove to do items using Language Understanding',
+    index: 4,
+  },
+  RespondingWithCardsSample: {
+    name: 'Responding with Cards',
+    description: 'A sample bot that uses Language Generation to create cards.',
+  },
+  AskingQuestionsSample: {
+    name: 'Asking Questions',
+    description: 'A sample bot that shows how to ask questions and capture user input.',
+  },
+  InterruptionSample: {
+    name: 'Interruptions',
+    description: 'An advanced sample bot that shows how to handle context switching and interruption in a conversation.',
+  },
+  RespondingWithTextSample: {
+    name: 'Responding with Text',
+    description: 'A sample bot that uses Language Generation to create bot responses.',
+  },
+  ControllingConversationFlowSample: {
+    name: 'Controlling Conversation Flow',
+    description: 'A sample bot that shows how to control the flow of a conversation.',
+  },
+  ActionsSample: {
+    name: 'Dialog Actions',
+    description: 'A sample bot that shows how to use Dialog Actions.',
+  },
+  QnAMakerLUISSample: {
+    name: 'QnA Maker and LUIS',
+    description: 'A sample bot that demonstrates use of both QnA Maker & LUIS',
+  }
+}
+
+
+function getSamples(): any[] {
+  const subPaths = fs.readdirSync(samplesDir);
+  const samples = [];
+  for (const subPath of subPaths) {
+    const fullPath = samplesDir + "/" + subPath;
+    if (!fs.statSync(fullPath).isDirectory()) {
+      continue;
+    }
+
+    // only looking for directories
+    const dirname = subPath;
+    let sample = { id: dirname, name: dirname, description: dirname, path: fullPath, ...samplesRegitry['*'] };
+    if (samplesRegitry[sample.id]) {
+      sample = { ...sample, ...samplesRegitry[sample.id] };
+    }
+    samples.push(sample);
+  }
+  samples.sort((a, b) => {
+    if (a.index && b.index) {
+      return a.index - b.index;
+    }
+    if (a.index) {
+      return -1;
+    }
+    return 1;
+  })
+  return samples;
+}
+
+function getBiolerPlates() {
+  return [
+    {
+      id: "biolerplate",
+      name: "biolerplate",
+      description: "base template for every bot template",
+      path: biolerplateDir,
+      tags: ["biolerplate"],
+      support: ["*"],
+    }
+  ]
+}
+
+const samples = getSamples();
+const biolerplates = getBiolerPlates();
+
+export default async (composer: any): Promise<void> => {
```

Comment:
```suggestion
export default async (composer: ComposerPluginRegistration): Promise<void> => {
```

Reply:
When i try to refer to pluginLoader package as dependencies, somehow the samples plugin can't build, with this error https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40905, seems related to types/express-server-core-static, i will see i can fix it today, if not, probably i will keep use any.

Reply:
tracked it in another issue. https://github.com/microsoft/BotFramework-Composer/issues/2912